regression/iterate_data.py

```diff
@@ -108,14 +108,23 @@ def main(args: Namespace, kwargs: dict[str, str]) -> None:
             start_time = time.time()
         for batch in dataloader:
             if args.sample_order_file is not None:
-                samples = [int(sample) for sample in batch['id']]
-                obj_gather_list = [
-                    torch.zeros(len(samples), dtype=torch.int64).cuda(get_rank())
-                    for _ in range(get_world_size())
-                ]
-                all_gather(obj_gather_list,
-                           torch.Tensor(samples).to(dtype=torch.int64).cuda(get_rank()))
-                barrier()
+                key = None
+                if 'id' in batch:
```

Comment:
Can we use `iterate_data.py` file with `imagedataset` ?

Reply:
We do sample order comparison and I'm trying to write the least amount of data to the files.